server/src/routers/MenuItemsRouter/parseMenuItemData.js

```diff
@@ -30,24 +30,21 @@ export default function parseMenuItemData(nutritionData, filterData, ingredientD
 
     // Process ingredient list
 	var ingredientList = [];
-	for (var i = 0; i < ingredientData.DATA.length; i++){
-		ingredientList.push(ingredientData.DATA[i][1]);
+	for (var ingredient of ingredientData.DATA){
+		ingredientList.push(ingredient[1]);
 	}
 	
 
 	// Process filters 
 	var filterList = []; // list of applicable filters
-	for(var i = 0; i < filters.length; i++){
-		if(filterData.DATA[0][i+2] == 1) {
-			filterList.push(filters[i]);
-		}
-	}
+	var boolFilters = filterData.DATA[0].slice(2, filters.length+2);
+    filters.map(function(filter, i) {if(boolFilters[i] == 1) filterList.push(filter)}.bind(this));
```

Comment:
More nit-picks -- if you use an arrow function, it'll look cleaner and you won't have to bind `this`!

Reply:
Sure, changed